domains/transfer/Transfer.API/API/Claiming/Automation/ClaimService.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using API.Claiming.Api.Repositories;
+using API.Claiming.Automation.Services;
+using Microsoft.Extensions.Logging;
+using ProjectOrigin.WalletSystem.V1;
+
+namespace API.Claiming.Automation;
+
+public class ClaimService : IClaimService
+{
+    private readonly ILogger<ClaimService> logger;
+    private readonly IClaimRepository claimRepository;
+    private readonly IProjectOriginWalletService walletService;
+
+    public ClaimService(ILogger<ClaimService> logger, IClaimRepository claimRepository, IProjectOriginWalletService walletService)
+    {
+        this.logger = logger;
+        this.claimRepository = claimRepository;
+        this.walletService = walletService;
+    }
+
+    public async Task Run(CancellationToken stoppingToken)
+    {
+        while (!stoppingToken.IsCancellationRequested)
+        {
+            logger.LogInformation("ClaimService running at: {time}", DateTimeOffset.Now);
+            try
+            {
+                var claimSubjects = await claimRepository.GetClaimSubjects();
+                foreach (var subjectId in claimSubjects.Select(x => x.SubjectId))
```

Comment:
Reading this I thought: "Does this run multiple times for the same SubjectId, if it is repeated in `claimSubject`"?

You could use .Distinct() in stead of .Select(). Since SubjectId is the key, it will give the same result, but reading this will be easier to see that this is only unique SubjectIds.

Reply:
I can't get it to look as elegant using destinct. Can you show me what you mean?

Reply:
claimSubjects.Select(x => x.SubjectId).Distinct()

Reply:
Sorry, you can also use .DistinctBy(). See https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/set-operations?branch=main#distinct-and-distinctby

Reply:
Yes but then I get ClaimSubjects and have to Select afterswards anyway